From 3f7f937b7bcabb5bb2fe544e75e6214ccabb137c Mon Sep 17 00:00:00 2001
From: Behan Webster <behanw@converseincode.com>
Date: Wed, 21 May 2014 12:02:30 -0700
Subject: [PATCH] kbuild, LLVMLinux: Add better clang cross build support

Add cross target to CC if using clang. Also add custom gcc toolchain path for
fallback gcc tools.

Signed-off-by: Behan Webster <behanw@converseincode.com>
---
 Makefile | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/Makefile b/Makefile
index 9fdfdd3..f2950c5 100644
--- a/Makefile
+++ b/Makefile
@@ -349,7 +349,16 @@ include $(srctree)/scripts/Kbuild.include
 
 AS		= $(CROSS_COMPILE)as
 LD		= $(CROSS_COMPILE)ld
+ifeq ($(COMPILER),clang)
+ifneq ($(CROSS_COMPILE),)
+REAL_CC		:= $(CC) -target $(CROSS_COMPILE:%-=%)
+endif
+ifneq ($(GCC_TOOLCHAIN),)
+REAL_CC		+= -gcc-toolchain $(GCC_TOOLCHAIN)
+endif
+else
 REAL_CC		= $(CROSS_COMPILE)gcc
+endif
 CPP		= $(CC) -E
 AR		= $(CROSS_COMPILE)ar
 NM		= $(CROSS_COMPILE)nm
-- 
1.9.1

